internal/pkg/deploy/cloudformation/stack/transformers.go

```diff
@@ -1039,13 +1039,24 @@ func convertSecrets(secrets map[string]manifest.Secret) map[string]template.Secr
 	if len(secrets) == 0 {
 		return nil
 	}
-	m := make(map[string]template.Secret)
+	m := make(map[string]template.Secret, len(secrets))
 	for name, mftSecret := range secrets {
-		var tplSecret template.Secret = template.SecretFromSSMOrARN(mftSecret.Value())
+		var tplSecret template.Secret
 		if mftSecret.IsSecretsManagerName() {
-			tplSecret = template.SecretFromSecretsManager(mftSecret.Value())
+			if mftSecret.SecretsManagerRequiresImport() {
```

Comment:
This is not a super useful case I think! Here is my reasonings. If user create a secret in a stack (which could either be an env addon, or just an arbitrary stack that they manage on their own),they most likely export the secret like this:
```yaml
Outputs:
  MySecret:
     Value: !Ref MySecret
     Export:
       Name: !Sub ${AWS::StackName}-MySecret
```

The [`!Ref`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-secretsmanager-secret.html#aws-resource-secretsmanager-secret-return-values) is going to have the ARN of the secret, not the name.  From the same [doc](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-secretsmanager-secret.html#aws-resource-secretsmanager-secret-return-values), we know that for secrets, you can only use `!Ref` which returns ARN, i.e. you can't `!GetAtt MySecret.Name`. 

Therefore, chances are, people don't export a secret's name, but a secret ARN. 

On the other hand, `secretsmanager` accepts the **name** of a secret, not the  **arn**. Therefore, `secretsmanager.from_cfn` isn't super useful.

I'd say we only need to take care of the following cases:
```yaml
secrets:
  MYSECRETIMPORTED:
    from_cfn: MYSTACK-secretsExportName
  MYSECRETPLAIN: 'arn'
  MYSECRETSMANAGER:
    secretsmanager: 'secretname'
  
```

Reply:
As discussed removed the implementation of `from_cfn` for `secretsmanger` that takes the name of a secret.

Reply:
I agreed it makes little sense to have `from_cfn` for `secretsmanager` given the fact that "secretsmanager only accepts the name of a secret, not the arn". I wonder if it is a problem we need to keep in mind since ideally both name and arn should work. It opens a door for using any cross-account secret I guess and leverages the benefit for having `from_cfn` 🤔